### PR TITLE
use odb traversal instead of tinkerpop api

### PR DESCRIPTION
**Pull request by mpollmeier**
important: this is part of the 'big bang' and should only be merged once we're happy with the migration guide and ready to merge all related PRs

* depends on https://github.com/ShiftLeftSecurity/overflowdb/pull/149
* depends on https://github.com/ShiftLeftSecurity/overflowdb-codegen/pull/24

Dependency tree *before* this PR:
deps

And after this PR:
deps

### CODE REVIEW

**Review by fabsx00 — APPROVED** [review]

**Review by hubertp — COMMENTED** [review]
nits

File: cpg.gron [hunk 1]
```diff
@@ -0,0 +1,4451 @@
+json = {};
```
[the remark this answers is not in the document]
Comment (mpollmeier) [reply cpg.gron:1]: whoops, thanks!

File: cpgvalidator/src/main/scala/io/shiftleft/cpgvalidator/validators/KeysValidator.scala [hunk 2]
```diff
@@ -38,8 +38,8 @@ class KeysValidator(errorRegistry: ValidationErrorRegistry) extends Validator {
   }
 
   private def validateCardinalityOne(dstNode: Node, nodeKeyType: String): Unit = {
-    val property = dstNode.propertyOption(PropertyKey(nodeKeyType))
-    if (property.isEmpty) {
+    val property = dstNode.propertyOption(new PropertyKey[Any](nodeKeyType))
+    if (!property.isPresent) {
```
[the remark this answers is not in the document]
Comment (mpollmeier) [reply cpgvalidator/src/main/scala/io/shiftleft/cpgvalidator/validators/KeysValidator.scala:42]: i agree, but propertyOption returns a java Optional (overflowdb-core doesn't have a scala dependency). 
we could implicitly convert in this (and similar) cases, but I figured it's not worth it

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/Tag.scala [hunk 3]
```diff
@@ -1,92 +1,98 @@
...
-        .order(By((x: Node) => x.id2))
-        .cast[nodes.Member]
-    )
+  def member: Traversal[nodes.Member] =
+    traversal
+      .in(EdgeTypes.TAGGED_BY)
+      .hasLabel(NodeTypes.MEMBER)
+      .l
+      .sortBy(_.id)
+      .to(Traversal) //TODO MP add orderBy/sortBy step
```
[the remark this answers is not in the document]
Comment (mpollmeier) [reply semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/Tag.scala:14]: yes. it's on my list of many small improvements once the big bang is in

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/types/expressions/generalizations/AstNode.scala [hunk 4]
```diff
@@ -1,216 +1,216 @@
...
+  def astMinusRoot: Traversal[nodes.AstNode] =
+    traversal.repeat(_.out(EdgeTypes.AST))(_.emitAllButFirst).cast[nodes.AstNode]
 
   /**
     * Direct children of node in the AST. Siblings are ordered by their `order` fields
     * */
-  def astChildren: NodeSteps[nodes.AstNode] =
-    new NodeSteps(raw.flatMap(_._astOut().asScala.toList.start.cast[nodes.AstNode].orderBy(_.order).raw))
+  def astChildren: Traversal[nodes.AstNode] =
+    // TODO MP add orderBy step to avoid conversion list/traversal
```
[the remark this answers is not in the document]
Comment (mpollmeier) [reply semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/types/expressions/generalizations/AstNode.scala:49]: same as above

**Pull request merged**